src/main/java/uk/gov/hmcts/reform/prl/services/caseaccess/RestrictedCaseAccessService.java

```diff
@@ -0,0 +1,217 @@
+        }
+
+        if (MARK_CASE_AS_RESTRICTED.equals(caseEvent)) {
+            applicantCaseName =  applicantCaseName + RESTRICTED_CASE;
+        } else if (MARK_CASE_AS_PRIVATE.equals(caseEvent)) {
+            applicantCaseName = applicantCaseName + PRIVATE_CASE;
+        }
+
+        caseDataUpdated.put(APPLICANT_CASE_NAME, applicantCaseName);
+        caseDataUpdated.put("caseNameHmctsInternal", applicantCaseName);
+        if (caseDataUpdated.containsKey(APPLICANT_OR_RESPONDENT_CASE_NAME)) {
+            caseDataUpdated.put(APPLICANT_OR_RESPONDENT_CASE_NAME, applicantCaseName);
+        }
+    }
+
+    public ResponseEntity<SubmittedCallbackResponse> changeCaseAccessRequestSubmitted(CallbackRequest callbackRequest) {
+        log.info("** restrictedCaseAccess event started");
+        Map<String, Object> caseDataUpdated = callbackRequest.getCaseDetails().getData();
+
+        log.info("** caseDataUpdated:: " + caseDataUpdated);
+        CaseSecurityClassificationEnum caseSecurityClassification
+            = CaseSecurityClassificationEnum.fromValue((String) caseDataUpdated.get(CASE_SECURITY_CLASSIFICATION));
+        log.info("CaseSecurityClassificationEnum::" + caseSecurityClassification);
+        StartAllTabsUpdateDataContent startAllTabsUpdateDataContent =
+            allTabService.getStartUpdateForSpecificEvent(
+                String.valueOf(callbackRequest.getCaseDetails().getId()),
+                CHANGE_CASE_ACCESS_AS_SYSUSER.getValue()
+            );
+        CaseDataContent caseDataContent = null;
+        switch (caseSecurityClassification) {
+            case RESTRICTED -> {
+                log.info("** inside restriced:: ");
+                caseDataContent = coreCaseDataService.createCaseDataContentOnlyWithSecurityClassification(
+                    startAllTabsUpdateDataContent.startEventResponse(),
+                    Classification.RESTRICTED
+                );
+            }
+            case PRIVATE -> {
+                log.info("** inside private:: ");
+                caseDataContent = coreCaseDataService.createCaseDataContentOnlyWithSecurityClassification(
+                    startAllTabsUpdateDataContent.startEventResponse(),
+                    Classification.PRIVATE
+                );
+            }
+            case PUBLIC -> {
+                log.info("** inside public:: ");
+                caseDataContent = coreCaseDataService.createCaseDataContentOnlyWithSecurityClassification(
+                    startAllTabsUpdateDataContent.startEventResponse(),
+                    Classification.PUBLIC
+                );
+            }
+            default -> {
+                log.info("** inside default:: ");
+                coreCaseDataService.createCaseDataContentOnlyWithSecurityClassification(
+                    startAllTabsUpdateDataContent.startEventResponse(),
+                    Classification.PUBLIC
+                );
+            }
+        }
+        coreCaseDataService.submitUpdate(
```

Comment:
For default, caseDataContent will be null, do we need to initialize it for some default value